presto-main/src/main/java/com/facebook/presto/execution/scheduler/group/DynamicBucketNodeMap.java

```diff
@@ -32,24 +32,24 @@
         extends BucketNodeMap
 {
     private final int bucketCount;
-    private final Int2ObjectMap<InternalNode> bucketToNode = new Int2ObjectOpenHashMap<>();
+    private final Int2ObjectMap<InternalNodeAndCache> bucketToNodeAndCache = new Int2ObjectOpenHashMap<>();
     private final boolean initialMapExist;
 
-    public DynamicBucketNodeMap(ToIntFunction<Split> splitToBucket, int bucketCount)
+    public DynamicBucketNodeMap(ToIntFunction<Split> splitToBucket, int bucketCount, boolean cacheable)
```

Comment:
It seems that this `cacheable` solely depends on whether there is `bucketToPreferredNode` in the argument? Is there a use case where we called the other constructor but set `cacheable` to `false`?